routes/courses.js

```diff
@@ -24,17 +24,17 @@ router.get('/new', function (req, res) {
 });
 
 router.post('/', (req, res) => {
-    UserModel.findById(req.body.userId)
+    UserModel.findById(req.body.userId).exec()
     .then(user => {
         const courseData = req.body.course;
-        courseData.user = user
+        courseData.user = user;
         return CourseModel.create(courseData);
     }).then(course => {
         res.redirect('/users/' + String(course.user._id));
     }).catch(err => {
         console.log(err);
-        res.redirect('/courses/new');
+        res.redirect('courses/new');
```

Comment:
I think this should stay as /courses/new because it's a url redirect, not a template

not sure if it matters